src/main/java/org/broadinstitute/hellbender/utils/codecs/GENCODE/GencodeGtfFeature.java

```diff
@@ -0,0 +1,1320 @@
+     * @return true if {@code other} is contained within the bounds of this {@link GencodeGtfFeature}, false otherwise.
+     */
+    public boolean contains(GencodeGtfFeature other) {
+        return (other.getStart() >= getStart()) && (other.getEnd() <= getEnd());
+    }
+
+    public void setFeatureOrderNumber(long featureOrderNumber) {
+        this.featureOrderNumber = featureOrderNumber;
+    }
+
+    // ================================================================================================
+
+    static public class OptionalField<T> {
+
+        private String name;
+        private T value;
+
+        public OptionalField(String name, T value) {
+            this.name = name;
+            this.value = value;
+        }
+
+        public String getName() {
+            return name;
+        }
+
+        public void setName(String name) {
+            this.name = name;
+        }
+
+        public T getValue() {
+            return value;
+        }
+
+        public void setValue(T value) {
+            this.value = value;
+        }
+
+        @Override
+        public String toString() {
+
+            StringBuilder sb = new StringBuilder();
+
+            sb.append(name);
+            sb.append(" ");
+
+            // We need to do some formatting for the numbers / non-numbers in the field:
+            String valueString = value.toString();
+            if ( valueString.matches("\\d\\d*") ) {
+                sb.append(valueString);
+                sb.append(";");
+            }
+            else {
+                sb.append("\"");
+                sb.append(valueString);
+                sb.append("\";");
+            }
+
+            return sb.toString();
+        }
+
+        @Override
+        public int hashCode() {
+            return this.toString().hashCode();
+        }
+
+        @Override
+        public boolean equals(Object other) {
+
+            if ( !(other instanceof OptionalField) ) {
+                return false;
+            }
+
+            OptionalField<?> otherOptionalField = (OptionalField<?>) other;
+
+            return (name.equals(otherOptionalField.name)) &&
+                    (value.equals(otherOptionalField.value));
+        }
+    }
+
+    // ================================================================================================
+
+    public enum AnnotationSource {
+        ENSEMBL,
+        HAVANA
+    }
+
+    public enum FeatureType {
+        gene("gene"),
+        transcript("transcript"),
+        selenocysteine("Selenocysteine"),
+        exon("exon"),
+        cds("CDS"),
+        start_codon("start_codon"),
+        stop_codon("stop_codon"),
+        utr("UTR");
+
+        private String serialized;
+
+        FeatureType(String serializedValue) { serialized = serializedValue; }
+
+        @Override
+        public String toString() { return serialized; }
+
+        public static FeatureType getEnum(String s) {
```

Comment:
No... the problem with the enums is that the valueOf will only work if the name of the enum is exactly the same as the string representation.  With some refactoring of the enums, this is no longer the case for any enum type, so this still has to be there.  On the plus side, I've now used the same pattern for all enums so they're at least uniform.